#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.1854974048244506 0.14840459729589528 0.971373660495711 1.4522959705408307
  position 0.10986806680967151 -1.144435498026873 0.5973441761546213
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 0 1 0 4.692820414042842e-06
  floorSize 10 10
}
DEF robot Robot {
  rotation 0 0 1 3.14
  children [
    RangeFinder {
      translation -0.05 7.96325e-05 0.03
      rotation 0 0 1 3.14
      width 320
      height 240
    }
    Emitter {
    }
    InertialUnit {
      name "imu"
    }
    Solid {
      translation -0.05 1.3268e-07 0.03
      rotation 0 0 1 3.14159
      children [
        Pose {
          rotation 1 0 0 -1.5708
          children [
            DEF BOARD Group {
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColorMap ImageTexture {
                      url [
                        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/nvidia/jetbot/protos/textures/raspberry_pi_camera.jpg"
                      ]
                    }
                    roughness 1
                    metalness 0
                  }
                  geometry IndexedFaceSet {
                    coord Coordinate {
                      point [
                        -0.0005 -0.01199995592153876 -0.011999999999919044
                        -0.0005 0.01200004407846124 -0.011999999999919044
                        -0.0005 -0.01200004407846124 0.011999999999919044
                        -0.0005 0.01199995592153876 0.011999999999919044
                        0.0005 -0.01199995592153876 -0.011999999999919044
                        0.0005 0.01200004407846124 -0.011999999999919044
                        0.0005 -0.01200004407846124 0.011999999999919044
                        0.0005 0.01199995592153876 0.011999999999919044
                      ]
                    }
                    texCoord TextureCoordinate {
                      point [
                        0 0
                        1 0
                        0 1
                        1 1
                        0.0001 0.0001
                        0.9999 0.0001
                        0.0001 0.9999
                        0.9999 0.9999
                      ]
                    }
                    coordIndex [
                      1, 0, 2, 3, -1, 4, 5, 7, 6, -1, 5, 1, 3, 7, -1, 0, 4, 6, 2, -1, 7, 3, 2, 6, -1, 4, 0, 1, 5, -1
                    ]
                    texCoordIndex [
                      1, 0, 2, 3, -1, 0, 1, 3, 2, -1, 1, 5, 7, 3, -1, 0, 4, 6, 2, -1, 7, 3, 2, 6, -1, 0, 4, 5, 1, -1
                    ]
                  }
                }
              ]
            }
          ]
        }
        Pose {
          translation -0.003 0 -0.0015
          children [
            Shape {
              appearance DEF LIGHT_GRAY_APPEARANCE PBRAppearance {
                baseColor 0.729412 0.741176 0.713725
                roughness 0.5
                metalness 0.2
              }
              geometry Box {
                size 0.004 0.022 0.027
              }
            }
          ]
        }
        Camera {
          width 320
          height 240
          recognition Recognition {
          }
        }
      ]
      name "CameraBody"
    }
    Gyro {
    }
    Accelerometer {
    }
    GPS {
    }
    Group {
    }
    DEF BODY Group {
      children [
        DEF LEFT_TRACK Pose {
          translation 0 0.0295 0.014
          rotation 1 0 0 -1.57
          children [
            DEF TRACK Group {
              children [
                Pose {
                  translation -0.049 0 0
                  children [
                    DEF TRACK_EXTRUSION Shape {
                      appearance DEF BLACK PBRAppearance {
                        baseColor 0 0 0
                        roughness 0.7
                        metalness 0
                      }
                      geometry IndexedFaceSet {
                        coord Coordinate {
                          point [
                            0 0.014 0
                            -0.007 0.01212 0
                            -0.01212 0.007 0
                            -0.014 0 0
                            -0.01212 -0.007 0
                            -0.007 -0.01212 0
                            0 -0.014 0
                            0 0.014 0
                            0 0.014 0.0198
                            -0.007 0.01212 0.0198
                            -0.01212 0.007 0.0198
                            -0.014 0 0.0198
                            -0.01212 -0.007 0.0198
                            -0.007 -0.01212 0.0198
                            0 -0.014 0.0198
                            0 0.014 0.0198
                          ]
                        }
                        coordIndex [
                          7, 6, 5, 4, 3, 2, 1, 0, -1, 8, 9, 10, 11, 12, 13, 14, 15, -1, 0, 1, 9, 8, -1, 1, 2, 10, 9, -1, 2, 3, 11, 10, -1, 3, 4, 12, 11, -1, 4, 5, 13, 12, -1, 5, 6, 14, 13, -1, 6, 7, 15, 14, -1
                        ]
                      }
                    }
                  ]
                }
                Pose {
                  translation 0.049 0 0
                  rotation 0 0 1 3.14
                  children [
                    DEF TRACK_EXTRUSION Shape {
                      appearance USE BLACK
                      geometry IndexedFaceSet {
                        coord Coordinate {
                          point [
                            0 0.014 0
                            -0.007 0.01212 0
                            -0.01212 0.007 0
                            -0.014 0 0
                            -0.01212 -0.007 0
                            -0.007 -0.01212 0
                            0 -0.014 0
                            0 0.014 0
                            0 0.014 0.0198
                            -0.007 0.01212 0.0198
                            -0.01212 0.007 0.0198
                            -0.014 0 0.0198
                            -0.01212 -0.007 0.0198
                            -0.007 -0.01212 0.0198
                            0 -0.014 0.0198
                            0 0.014 0.0198
                          ]
                        }
                        coordIndex [
                          7, 6, 5, 4, 3, 2, 1, 0, -1, 8, 9, 10, 11, 12, 13, 14, 15, -1, 0, 1, 9, 8, -1, 1, 2, 10, 9, -1, 2, 3, 11, 10, -1, 3, 4, 12, 11, -1, 4, 5, 13, 12, -1, 5, 6, 14, 13, -1, 6, 7, 15, 14, -1
                        ]
                      }
                    }
                  ]
                }
                Pose {
                  translation 0.049 0 0.0101
                  children [
                    DEF TRACK_WHEEL Shape {
                      appearance DEF GRAY PBRAppearance {
                        roughness 0.2
                      }
                      geometry Cylinder {
                        height 0.0205
                        radius 0.011
                      }
                    }
                  ]
                }
                Pose {
                  translation -0.049 0 0.0101
                  children [
                    USE TRACK_WHEEL
                  ]
                }
                Pose {
                  translation 0 -0.0125 0.0099
                  children [
                    DEF TRACK_BOX Shape {
                      appearance USE BLACK
                      geometry Box {
                        size 0.098 0.003 0.0198
                      }
                    }
                  ]
                }
                Pose {
                  translation 0 0.0125 0.0099
                  children [
                    USE TRACK_BOX
                  ]
                }
              ]
            }
          ]
        }
        DEF RIGHT_TRACK Pose {
          translation 0 -0.0295 0.014
          rotation 1 0 0 1.57
          children [
            USE TRACK
          ]
        }
        DEF BASE Pose {
          translation 0 0 0.014
          children [
            Pose {
              translation -0.047 0 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.721569 1 0.0745098
                  }
                  geometry Box {
                    size 0.022 0.056 0.014
                  }
                }
              ]
            }
            Pose {
              translation 0.048 0 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.666667 0.666667 1
                  }
                  geometry Box {
                    size 0.016 0.056 0.014
                  }
                }
              ]
            }
            Pose {
              translation 0.002 0 0
              children [
                Shape {
                  appearance USE GRAY
                  geometry Box {
                    size 0.0765 0.043 0.014
                  }
                }
              ]
            }
            Pose {
              translation -0.049 0 0
              rotation 1 0 0 1.57
              children [
                DEF WHEEL_AXE Shape {
                  appearance USE GRAY
                  geometry Cylinder {
                    height 0.059
                    radius 0.00635
                  }
                }
              ]
            }
            Pose {
              translation 0.049 0 0
              rotation 1 0 0 1.57
              children [
                USE WHEEL_AXE
              ]
            }
          ]
        }
      ]
    }
    DEF FIRST_LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0.049 0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor00"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor00"
        }
      ]
      endPoint Solid {
        translation 0.04899994958950295 0.039399977476371965 0.013999639355855048
        rotation -0.9754070069033555 1.1968589451732343e-07 0.2204113674106281 6.283179238033401
        name "first left wheel"
        contactMaterial "wheel"
        boundingObject DEF BOUNDING_WHEEL Pose {
          rotation 1 0 0 -1.57
          children [
            Cylinder {
              height 0.0198
              radius 0.014
            }
          ]
        }
        physics DEF PHYSICS_FRONT_WHEEL Physics {
          density -1
          mass 0.005
        }
      }
    }
    DEF SECOND_LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0.0245 0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor01"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor01"
        }
      ]
      endPoint Solid {
        translation 0.02449993019724922 0.03939998879767114 0.013999589950905061
        rotation -0.9753926443387754 1.2011716654897612e-07 0.22047491777931758 6.2831792383260945
        name "second left wheel"
        contactMaterial "wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_FRONT_WHEEL
      }
    }
    DEF THIRD_LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0 0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor02"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor02"
        }
      ]
      endPoint Solid {
        translation -8.919979931826062e-08 0.03940000011896968 0.013999540546717603
        rotation -0.9753782769936318 1.1967033440166938e-07 0.2205384700430285 6.283179238691981
        name "third left wheel"
        contactMaterial "center_wheel"
        boundingObject USE BOUNDING_WHEEL
        physics DEF PHYSICS_CENTER_WHEEL Physics {
          density -1
          mass 0.005
        }
      }
    }
    DEF FOURTH_LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor -0.0245 0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor03"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor03"
        }
      ]
      endPoint Solid {
        translation -0.024500108601943252 0.039400011440267564 0.013999491143339832
        rotation -0.9753639038465213 1.2003304037973582e-07 0.22060202871519458 6.283179238984706
        name "fourth left wheel"
        contactMaterial "wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_FRONT_WHEEL
      }
    }
    DEF LAST_LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor -0.049 0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor04"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor04"
        }
      ]
      endPoint Solid {
        translation -0.04900012800408704 0.03940002276156519 0.013999441739961881
        rotation -0.975349525727492 1.198102924588975e-07 0.22066559012030404 6.28317923920426
        name "last left wheel"
        contactMaterial "wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_FRONT_WHEEL
      }
    }
    DEF FIRST_RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0.049 -0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor05"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor05"
        }
      ]
      endPoint Solid {
        translation 0.048999971601092644 -0.039400022954759834 0.013999553272158722
        rotation -0.9753761590155398 1.1976615798412383e-07 0.22054783704239317 6.283179237521221
        name "first right wheel"
        contactMaterial "wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_FRONT_WHEEL
      }
    }
    DEF SECOND_RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0.0245 -0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor06"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor06"
        }
      ]
      endPoint Solid {
        translation 0.024499950521821155 -0.03940001146122501 0.013999504135168544
        rotation -0.975361989834043 1.1998259316675122e-07 0.22061049110811118 6.28317923781389
        name "second right wheel"
        contactMaterial "wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_FRONT_WHEEL
      }
    }
    DEF THIRD_RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0 -0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor07"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor07"
        }
      ]
      endPoint Solid {
        translation -7.070635438516768e-08 -0.03939999995177826 0.013999455020936332
        rotation -0.9753478267178387 1.1976003672465408e-07 0.22067309966730092 6.283179238033401
        name "third right wheel"
        contactMaterial "center_wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_CENTER_WHEEL
      }
    }
    DEF FOURTH_RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor -0.0245 -0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor08"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor08"
        }
      ]
      endPoint Solid {
        translation -0.024500091596166982 -0.039399988630512014 0.013999405854081626
        rotation -0.9753336951591044 1.2003592011682004e-07 0.22073555012111848 6.28317923839927
        name "forth right wheel"
        contactMaterial "wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_FRONT_WHEEL
      }
    }
    DEF LAST_RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor -0.049 -0.0394 0.014
      }
      device [
        RotationalMotor {
          name "wheel_motor09"
          maxVelocity 50
        }
        PositionSensor {
          name "wheel_sensor09"
        }
      ]
      endPoint Solid {
        translation -0.04900011214651826 -0.039399977309281235 0.013999356633256457
        rotation -0.9753195908103611 1.1967147047139572e-07 0.22079786181368535 6.283179238691981
        name "last right wheel"
        contactMaterial "wheel"
        boundingObject USE BOUNDING_WHEEL
        physics USE PHYSICS_FRONT_WHEEL
      }
    }
  ]
  model "SRV-1"
  boundingObject Group {
    children [
      Pose {
        translation 0 0 0.03
        children [
          Box {
            size 0.114 0.0535 0.045
          }
        ]
      }
      Pose {
        translation -0.0039 0 0.063
      }
    ]
  }
  physics Physics {
    density -1
    mass 0.324
    centerOfMass [
      -0.005 0 0.028
    ]
    inertiaMatrix [
      0.000267079 0.000473553 0.000418585
      0 -6.02166e-06 0
    ]
  }
  controller "<extern>"
  supervisor TRUE
}
Solid {
  translation 0.43 0 0.0325
  children [
    DEF BALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 1 0.54 0.08
        roughness 0.3
        metalness 0
      }
      geometry Sphere {
        radius 0.01
        subdivision 4
      }
    }
  ]
  name "ball"
  model "ball"
  boundingObject USE BALL_SHAPE
  physics Physics {
    density -1
    mass 0.055
    centerOfMass [
      0 0 -0.0001
    ]
    inertiaMatrix [
      2.3e-05 2.3e-05 2.3e-05
      0 0 0
    ]
    damping Damping {
      linear 0.17
      angular 0.33
    }
  }
  recognitionColors [
    0 0 0
  ]
}
Robot {
  children [
    Receiver {
    }
  ]
  name "supervisor"
  controller "supervisor"
  supervisor TRUE
}
